spinalcordtoolbox/scripts/sct_label_vertebrae.py

```diff
@@ -347,47 +343,39 @@ def main(argv=None):
         label_vert('segmentation_straight.nii', 'labeldisc_straight.nii.gz', verbose=1)
 
     else:
-        # create label to identify disc
         printv('\nCreate label to identify disc...', verbose)
-        fname_labelz = os.path.join(path_tmp, file_labelz)
-        if initz or initcenter:
-            if initcenter:
-                # find z centered in FOV
-                nii = Image('segmentation.nii').change_orientation("RPI")
-                nx, ny, nz, nt, px, py, pz, pt = nii.dim  # Get dimensions
-                z_center = int(np.round(nz / 2))  # get z_center
-                initz = [z_center, initcenter]
-
-            im_label = create_labels_along_segmentation(Image('segmentation.nii'), [(initz[0], initz[1])])
-            im_label.data = dilate(im_label.data, 3, 'ball')
+        fname_labelz = os.path.join(path_tmp, 'labelz.nii.gz')
+        if initcenter is not None:
+            # find z centered in FOV
+            nii = Image('segmentation.nii').change_orientation("RPI")
+            nx, ny, nz, nt, px, py, pz, pt = nii.dim
+            z_center = round(nz / 2)
+            initz = [z_center, initcenter]
+        if initz is not None:
+            im_label = create_labels_along_segmentation(Image('segmentation.nii'), [tuple(initz)])
             im_label.save(fname_labelz)
-
-        elif fname_initlabel:
+        elif fname_initlabel is not None:
             Image(fname_initlabel).save(fname_labelz)
-
         else:
             # automatically finds C2-C3 disc
             im_data = Image('data.nii')
             im_seg = Image('segmentation.nii')
-            if not remove_temp_files:  # because verbose is here also used for keeping temp files
-                verbose_detect_c2c3 = 2
-            else:
-                verbose_detect_c2c3 = 0
+            # because verbose is also used for keeping temp files
+            verbose_detect_c2c3 = 0 if remove_temp_files else 2
             im_label_c2c3 = detect_c2c3(im_data, im_seg, contrast, verbose=verbose_detect_c2c3)
             ind_label = np.where(im_label_c2c3.data)
-            if not np.size(ind_label) == 0:
-                im_label_c2c3.data[ind_label] = 3
-            else:
+            if np.size(ind_label) == 0:
                 printv('Automatic C2-C3 detection failed. Please provide manual label with sct_label_utils', 1, 'error')
                 sys.exit()
+            im_label_c2c3.data[ind_label] = 3
```

Comment:
Minor thing I noticed while looking at this change, but we `sys.exit()` without passing an error code, which means we're [defaulting to zero](https://docs.python.org/3/library/sys.html#sys.exit). We might want to switch that to `sys.exit(1)`?

Reply:
Good catch! I'll fix it.